### ISSUE TITLE
Tab key escape character text width - bug?

### ISSUE THREAD

**Issue by FedeClaudi** [JuliaLang/julia#45377]
Hi, 
Thank you for the amazing work on Julia.

According to the docs for [`textwidth`](https://docs.julialang.org/en/v1/base/strings/#Base.Unicode.textwidth):
> Give the number of columns needed to print a character.

But:
```Julia
julia> textwidth('\t')
0

julia> print('\t'); print("next")
        next
julia> 
```

Perhaps I'm misunderstanding something here, but if `textwidth` gives the width of the character as printed in the terminal this seems to be wrong. 

It came up here: https://github.com/FedeClaudi/Term.jl/issues/92

---

```
julia> versioninfo()
Julia Version 1.7.1
Commit ac5cc99908 (2021-12-22 19:35 UTC)
Platform Info:
  OS: macOS (x86_64-apple-darwin19.5.0)
  CPU: Intel(R) Core(TM) i9-9880H CPU @ 2.30GHz
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-12.0.1 (ORCJIT, skylake)
Environment:
  JULIA_EDITOR = code
```

**Comment by elextr** [comment]
Tab doesn't have a width as such, it is a positioning command, it outputs one space then positions the next character at a multiple of the tab width set for the output device.  For example if you did:

```Julia
print("abc"); print('\t'); print("next");
```

the `"next"` would be in the same place, the tab would effectively be only 5 spaces wide.  

So `textwidth()` can't account for that with a fixed width, there is no sensible answer.  

The equivalent POSIX `wcwidth()` returns -1 to indicate no good answer, but the the `utf8proc_charwidth()` which underlies Julia `textwidth()` defines that it returns 0 where `wcwidth()` would return -1.

**Comment by FedeClaudi** [comment]
I see, that's fair then - thank you for the quick reply. 

What would be the best practice when one cares about the width of the output text in this case? Just discourage users from having `\t` in there?

**Comment by elextr** [comment]
I had a ready answer on tab because we recently looked at the problem in another project where we wanted to align positions on more than one line of "stdout".  We found no general answer, it can be made to work in some cases, but not generally.

1. Characteristics of the output device matter, if its not using a monospaced font all bets are off, no kerning, and relevant to this issue, tab settings need to be known, and in many cases these are unknown for stdout, eg when the device is hidden behind a pipe.
2. Apart from tab (which can be handled if you keep track of the current column) ASCII is generally ok
3. Unicode has code points whose glyphs are typically zero width, half width, full width and double width, some that combine with the preceeding character (eg diacriticals) ... unless they are separated by the misnamed Combining Graphene Joiner, and emojis with variation selectors, sigh, gave up!
